events/domEvents.js

```diff
@@ -35,6 +37,8 @@ const domEvents = () => {
     }
     // TODO: CLICK EVENT FOR VIEW BOOK DETAILS
     if (e.target.id.includes('view-book-btn')) {
+      const [, firebaseKey] = e.target.id.split('--');
+      getBookDetails(firebaseKey).then(viewBook);
```

Comment:
Check out this line. Are you passing viewbook anything? It's expecting something.